pulp_rpm/app/comps.py

```diff
@@ -1,4 +1,25 @@
 import hashlib
+import libcomps
+
+
+def list_to_idlist(lst):
+    """
+    Convert list to libcomps IdList object.
+
+    Args:
+        list: a list
+
+    Returns:
+        idlist: a libcomps IdList
+
+    """
+    idlist = libcomps.IdList()
+
+    for i in lst:
+        group_id = libcomps.GroupId(i['name'], i['default'])
+        idlist.append(group_id)
```

Comment:
The publication of this fails because in the remote source there is incorrect data, 2 groups of the same name/id are in one environment. http://mirror.linux.duke.edu/pub/centos/7/os/x86_64/.
To avoid that, you can eliminate duplicates:

```
    for i in lst:
        group_id = libcomps.GroupId(i['name'], i['default'])
        if group_id not in idlist:
            idlist.append(group_id)
```

Reply:
Thanks for spotting this!  Making the change to handle the duplicate data.